packages/sanity/src/core/form/studio/assetSource/DefaultSource.tsx

```diff
@@ -14,8 +14,51 @@ const PER_PAGE = 200
 const ASSET_TYPE_IMAGE = 'sanity.imageAsset'
 const ASSET_TYPE_FILE = 'sanity.fileAsset'
 
-const buildQuery = (start = 0, end = PER_PAGE, assetType = ASSET_TYPE_IMAGE) => `
-  *[_type == "${assetType}"] | order(_updatedAt desc) [${start}...${end}] {
+const buildFilterQuery = (acceptParam: string) => {
+  const WILDCARD_ACCEPT = ['image/*', 'audio/*', 'video/*']
+  const acceptItems = acceptParam.split(',').map((accept) => accept.trim())
+
+  const typesForFilter: {[key: string]: string} = acceptItems.reduce(
+    (acceptTypes: {[key: string]: string}, acceptR: string) => {
+      // builds the wildcard part of the groq query
+      if (WILDCARD_ACCEPT.includes(acceptR)) {
+        const wild = `mimeType match '${acceptR}' || ${acceptTypes.wildcards}`
+        return {...acceptTypes, ...{wildcards: wild}}
+      }
+
+      // builds the extension part of the groq query (and removes the .)
+      if (acceptR.indexOf('.') === 0) {
+        const exte = `'${acceptR.replace('.', '')}', ${acceptTypes.mimes}`
+        return {...acceptTypes, ...{extensions: exte}}
+      }
+
+      // all that remains is then the mime types, so we build that part
+      const mime = `'${acceptR}', ${acceptTypes.mimes}`
+      return {...acceptTypes, ...{mimes: mime}}
+    },
+    {mimes: '', extensions: '', wildcards: ''}
+  )
+
+  // when no accept filter is set, we don't need to add the filter condition
+  return `&&
```

Comment:
wondering if it makes sense to leave this `&&` out here and move it to the query instead - would make the function more future proof and reusable I believe

Reply:
The reason why I kept it here was because in my line of thinking if there is not accept params then it would not have a "loose" && in the query itself. Unless I'm misunderstanding your suggestion 😄